meta-mender-core/classes/mender-dataimg.bbclass

```diff
@@ -2,9 +2,22 @@
 # filesystem.
 
 IMAGE_CMD_dataimg() {
-    dd if=/dev/zero of="${WORKDIR}/data.${ARTIFACTIMG_FSTYPE}" count=0 bs=1M seek=${MENDER_DATA_PART_SIZE_MB}
-    mkfs.${ARTIFACTIMG_FSTYPE} -F "${WORKDIR}/data.${ARTIFACTIMG_FSTYPE}" -d "${IMAGE_ROOTFS}/data" -L data
-    install -m 0644 "${WORKDIR}/data.${ARTIFACTIMG_FSTYPE}" "${IMGDEPLOYDIR}/${IMAGE_NAME}.dataimg"
+    if [ ${MENDER_DATA_PART_FSTYPE_TO_GEN} = "btrfs" ]; then
+        force_flag="-f"
+        root_dir_flag="-r"
+    else #Assume ext3/4
+        force_flag="-F"
+        root_dir_flag="-d"
+    fi
+
+    dd if=/dev/zero of="${WORKDIR}/data.${MENDER_DATA_PART_FSTYPE_TO_GEN}" count=0 bs=1M seek=${MENDER_DATA_PART_SIZE_MB}
+    mkfs.${MENDER_DATA_PART_FSTYPE_TO_GEN} \
+        ${force_flag} \
+        "${WORKDIR}/data.${MENDER_DATA_PART_FSTYPE_TO_GEN}" \
+        ${root_dir_flag} "${IMAGE_ROOTFS}/data" \
```

Comment:
Didn't spot this the first time around: Use `$force_flag` notation for shell variables, not `${force_flag}`. Reduces chance that it will clash with Bitbake parsing.

Same goes for `$root_dir_flag`

Reply:
Got it. I'm guessing this is taking advantage of the fact that bitbake only recognizes variables with the {}?

Reply:
Correct!